catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/PhiroMoveMotorBackwardBrickTest.java

```diff
@@ -111,4 +118,53 @@ public void testPhiroMoveMotorBackwardBrick() {
 		onBrickAtPosition(brickPosition).onFormulaTextField(R.id.brick_phiro_motor_backward_action_speed_edit_text)
 				.checkShowsNumber(5);
 	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class, Cat.Gadgets.class})
+	@Test
+	public void testPhiroMoveMotorBackwardBrickUndo() {
+		int[] setSpeeds = {1, 2, 3};
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(brickPosition).checkShowsText(R.string.brick_phiro_motor_backward_action);
+
+		onBrickAtPosition(brickPosition).onChildView(withId(R.id.brick_phiro_motor_backward_action_speed_edit_text))
+				.perform(click());
+
+		onSingleSeekbar().performSwipePhiroSeekbar(setSpeeds[0] / SeekbarsPickerWrapper.MAX_PHIRO_SEEKBAR_VALUE)
+				.performSwipePhiroSeekbar(setSpeeds[1] / SeekbarsPickerWrapper.MAX_PHIRO_SEEKBAR_VALUE)
+				.performSwipePhiroSeekbar(setSpeeds[2] / SeekbarsPickerWrapper.MAX_PHIRO_SEEKBAR_VALUE);
+
+		onSingleSeekbar().performUndo()
+				.performUndo()
+				.performUndo();
+		onSingleSeekbar().pressBack(SeekbarsPickerWrapper.PHIRO_SEEKBAR_MATCHER);
+
+		onSingleSeekbar().discardChanges();
+
+		onBrickAtPosition(brickPosition).onFormulaTextField(R.id.brick_phiro_motor_backward_action_speed_edit_text)
+				.checkShowsNumber(initialSpeed);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class, Cat.Gadgets.class})
+	@Test
+	public void testPhiroMoveMotorBackwardBrickUndoRedo() {
+		int[] setSpeeds = {45, 10};
+
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(brickPosition).checkShowsText(R.string.brick_phiro_motor_backward_action);
+
+		onBrickAtPosition(brickPosition).onChildView(withId(R.id.brick_phiro_motor_backward_action_speed_edit_text))
+				.perform(click());
+
+		onSingleSeekbar().performSwipePhiroSeekbar(setSpeeds[0] / SeekbarsPickerWrapper.MAX_PHIRO_SEEKBAR_VALUE)
+				.performSwipePhiroSeekbar(setSpeeds[1] / SeekbarsPickerWrapper.MAX_PHIRO_SEEKBAR_VALUE);
+
+		onSingleSeekbar().performUndo();
+
+		onSingleSeekbar().pressBack(SeekbarsPickerWrapper.PHIRO_SEEKBAR_MATCHER);
```

Comment:
combine this and the save changes below in the wrapper. (also see the comment about pressBack i wrote down there)

saveChanges and discardChanges should do the backpress aswell, and you could rename them to "closeAndSave" or something similar.